src/dmd/lexer.d

```diff
@@ -141,6 +147,43 @@ class Lexer
         }
     }
 
+    version (DMDLIB)
+    {
+        this(const(char)* filename, const(char)* base, size_t begoffset, size_t endoffset,
+            bool doDocComment, bool commentToken, bool whitespaceToken)
+        {
+            this(filename, base, begoffset, endoffset, doDocComment, commentToken);
+            this.whitespaceToken = true;
+            this.popFront();
+        }
+
+        bool empty()
+        {
+            return front() == TOK.endOfFile;
+        }
+
+        TOK front() const @property
+        {
+            return token.value;
+        }
+
+        void popFront()
+        {
+            nextToken();
+        }
```

Comment:
The addition of a range interface should be done in another commit / PR

Reply:
Will do